### PR TITLE
Refactor and upgrade dashboard statistics

### PR DESCRIPTION
**Pull request by AdamKasp**
| Q               | A
| --------------- | -----
| Branch?         | master
| Bug fix?        | no
| New feature?    | yes
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | Related to #11371, closes #11405
| License         | MIT

### CODE REVIEW

File: src/Sylius/Bundle/CoreBundle/Doctrine/DQL/Year.php [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+declare(strict_types=1);
+
+namespace Sylius\Bundle\CoreBundle\Doctrine\DQL;
+
+use Doctrine\ORM\Query\AST\Functions\FunctionNode;
+use Doctrine\ORM\Query\Lexer;
+use Doctrine\ORM\Query\Parser;
+use Doctrine\ORM\Query\SqlWalker;
+
+final class Year extends FunctionNode
```
Comment (SirDomin) [inline src/Sylius/Bundle/CoreBundle/Doctrine/DQL/Year.php:21]: do we need 4 different classes only to return string based on class name ?

File: src/Sylius/Component/Core/Dashboard/IntervalsConverter.php [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Component\Core\Dashboard;
+
+final class IntervalsConverter implements IntervalsConverterInterface
+{
+    public function getIntervals(
```
Comment (GSadee) [inline src/Sylius/Component/Core/Dashboard/IntervalsConverter.php:18]:
```suggestion
    public function getInterval(
```
?

File: src/Sylius/Component/Core/Dashboard/SalesDataProvider.php [hunk 3]
```diff
@@ -15,49 +15,42 @@
...
+        \DateTimeInterface $endDate,
+        string $interval,
+        ChannelInterface $channel,
+        string $dateFormat
+    ): SalesSummaryInterface {
+        $provider = $this->salesSummaryProviderResolver->getSalesSummaryProvider($this->entityManager->getConnection());
+
+        $data = $provider->getSalesSummary($channel, $startDate, $endDate, $interval);
+
+        $dateIntervals = $this->intervalsConverter->getIntervals($startDate, $endDate, $interval);
```
Comment (GSadee) [inline src/Sylius/Component/Core/Dashboard/SalesDataProvider.php:52]: Should $dateIntervals be in the plural?

File: src/Sylius/Component/Core/Provider/MySQLSalesSummaryProvider.php [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+    public function __construct(OrderRepositoryInterface $orderRepository)
+    {
+        $this->orderRepository = $orderRepository;
+    }
+
+    /** @psalm-return array<int, int> Intervals to sales amount */
+    public function getSalesSummary(ChannelInterface $channel, \DateTimeInterface $startDate, \DateTimeInterface $endDate, string $interval
+    ): array
+    {
```
Comment (GSadee) [inline src/Sylius/Component/Core/Provider/MySQLSalesSummaryProvider.php:34]:
```suggestion
    public function getSalesSummary(
        ChannelInterface $channel, 
        \DateTimeInterface $startDate, 
        \DateTimeInterface $endDate, 
        string $interval
    ): array {
```

File: src/Sylius/Component/Core/Resolver/SalesSummaryProviderResolver.php [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+    {
+        $this->locator = $locator;
+    }
+
+    public function getSalesSummaryProvider(Connection $connection): SalesSummaryProviderInterface
+    {
+        $databaseName = $connection->getDatabasePlatform()->getName();
+
+        if (!$this->locator->has($databaseName)) {
+            return $this->locator->get('unHandled');
```
Comment (GSadee) [inline src/Sylius/Component/Core/Resolver/SalesSummaryProviderResolver.php:35]:
```suggestion
            return $this->locator->get('unhandled');
```

File: config/packages/doctrine.yaml [hunk 6]
```diff
@@ -10,5 +10,4 @@ doctrine:
         driver: 'pdo_mysql'
         server_version: '5.7'
         charset: UTF8
-
```
Comment (pamil) [inline config/packages/doctrine.yaml:13]: ❔

**Comment by pamil** [comment]
Rebase needed.

**Comment by GSadee** [comment]
Thanks, Adam! :1st_place_medal:

**Pull request merged**